Lightwallet support + mobile redesign (no merge)

Jaqueeee opened the pull request:
requires #2109

- Lightwallet support (mymonero/openmonero)
- reworked layout for smaller screens.
- Better remote node integration. Option to use remote node while syncing local node
- Restore/export wallet with QR code
- Moved seed&Keys to own tab. 
- other small changes and improvements

Some cleanup and optimisations is needed before this can be merged. PR'ing to get feedback from testers. 

**Experimental Android build**
https://www.dropbox.com/s/cv3ey6h93xlq4z9/monero-wallet-gui.apk?dl=0
Tested on Xiaomi Redmi 4. My goal is to make the interface as dynamic as possible. Would love to get feedback on how it looks on other screens. 
**NOTE:** this is not an official build. (Don't trust it)
If you want to build yourself I recommend these instructions from @MoroccanMalinois 
https://github.com/MoroccanMalinois/monero-core/tree/android_dockerfile/android

**Known issues**
- ~~Some screens are still little too wide for mobile, but all components should be reachable.~~
- LightWallet server is hadcoded to mymonero currently. If you wan't to use your own openmonero server you have to alter the code yourself. https://github.com/monero-project/monero-core/compare/master...Jaqueeee:openmonero?expand=1#diff-3d442cd229041f412bc48acdcee456bcR975
- MyMonero will stop syncing the wallet after some time. You have to login with your seed on mymonero.com to trigger the wallet sync if it lags behind. 
- Connecting to remote nodes doesn't work

Android
- ~~Selections handles can get stuck on screen after selecting text~~
- Translations doesn't work yet
- ~Crash when using backbutton~

IOS
- QR-code scanner not implemented yet

**Edit**: new APK link

mrowwy commented:
Password managers like LastPass do not detect/recognize the password field for "auto-fill". I'm am unsure exactly what would need to change to make them compatible.

Jaqueeee commented:
**Update 170625**
- Added workaround for the android text selection bug (https://bugreports.qt.io/browse/QTBUG-58700)
- redesigned some pages to fit smaller screens. 
- Updated the apk-binary

unknownids commented:
thanks for the work Jaqueeee :D

Heres some screens from the latest build (I assume you updated the dropbox link?) Anyway this is for the S6 Edge http://imgur.com/a/aIMIO

some comments 

crash: clicked QR code button, successfully opened camera, click the back arrow to get back into app and it crashed. 
reproducable?: yes

crash: monero at default send screen after logging in, hitting the back arrow on the s6 edge to get back to the android desktop crashes the app
reproducable?: yes

at the menu selection dragging the menu down and letting it bounce back up removes the yellow icon at the top left (see screenshots)
reproducable: yes

keep up the good work

MoroccanMalinois commented on src/libwalletqt/WalletManager.cpp:
```diff
@@ -284,8 +289,36 @@ QString WalletManager::resolveOpenAlias(const QString &address) const
-        return m_currentWallet->parse_uri(uri, address, payment_id, amount, tx_description, recipient_name, unknown_parameters, error);
+    if (m_currentWallet) {
+        m_currentWallet->parse_uri(uri, address, payment_id, amount, tx_description, recipient_name, unknown_parameters, error);
+        return true;
```
always true ?

SamsungGalaxyPlayer commented:
More screenshots from an OPO. Very similar to @unknownids S6 Edge: http://imgur.com/a/qV9hg

I replicated the back button crash. Hitting the back button at any time without the keyboard open crashes the application.

Jaqueeee commented:
apk updated again. 
- Fixed back button crash
- hide verify/sign on mobiles
- small redesign on transfer page

ping @SamsungGalaxyPlayer @medusadigital @unknownids @mrowwy @MoroccanMalinois

nixuxmr commented:
Looking that great on an OP2 too, thanks for this @Jaqueeee !
Can confirm, the back button crash is fixed also when returning from QR scan (updated apk).

SamsungGalaxyPlayer commented:
@Jaqueeee back button works now (see images) https://imgur.com/a/8YmRi

oakey22 commented:
When selecting the fee for a send transaction, you are unable to select the 1x fee once you select something else.

amiuhle commented:
QR Code scanning for an address works, but when it has an added amount, it doesn't get parsed correctly.

Example: `monero:A3Brqw9sVmwLyWS8EWeUw1VqpqfwnDHTkG7Pb4NJ3RmZWeeMZhGMe2ZXz4bSk7BbtEYF5981nLxkDYQ6B46tX5DMVoPGjfBxi6tLZwHK6E?tx_amount=0.6380155748871774`

amiuhle commented:
![screenshot_20170629-210004](https://user-images.githubusercontent.com/460091/27705496-5286e146-5d0e-11e7-8c8a-6e609110c990.png)

snaggen commented:
The 5X have the sensor flipped, so it requires applications to use the newer camera apis. Not sure exactly which api but there was a lot noise about this when the 5x came out, so it should be easy to google.

mrowwy commented:
-Text cutoff at bottom of seed indicating which wallet type is selected.
-Button coloring for selection state in settings is not intuitive to me. It seems like the more vibrant one should be selected.
-3-bar icon ( top-left ) disappears when in sidebar
-text cutoff during wallet creation process
-sometimes cannot paste password when in app. I have to leave the app and come back to see the paste option ( would be nice if LastPass could detect ;) )

http://imgur.com/a/H1nyH

jonathancross commented:
@amiuhle The "paid" [Barcode Scanner+ Simple](https://play.google.com/store/apps/details?id=com.srowen.bs.android.simple) fixed the flipped screen for me on Nexus 5x with Mycelium, so will probably work here as well.  The app is [Open Source](https://github.com/zxing/zxing), needs minimal permissions and the $1 you pay goes to WWF / Rainforest Alliance.

ReilySiegel commented:
Transaction will not send: Transaction cannot pay for itself.

This occurs when trying to send all XMR in a wallet (0.01 XMR) at x1 fees. When imported into the desktop GUI, the transaction sends fine.

```
Amount: 0.006327110000
Fee: 0.003672890000

Ringsize: 5
Number of transactions: 1
```

grigio commented:
Hi this is my feedback, i see very little label specially in the settings screen and I think the "audio recording permission" isn't needed

Jaqueeee commented:
Closing in favor of #826
